src/item/enchantment/EnchantmentHelper.php

```diff
@@ -0,0 +1,231 @@
+			$lastEnchantment = self::getRandomWeightedEnchantment($random, $availableEnchantments);
+			$resultEnchantments[] = $lastEnchantment;
+
+			// With probability (cost + 1) / 50, continue adding enchantments
+			while($random->nextFloat() <= ($cost + 1) / 50){
+				// Remove from the list of available enchantments anything that conflicts
+				// with previously-chosen enchantments
+				$availableEnchantments = array_filter(
+					$availableEnchantments,
+					fn(EnchantmentInstance $e) => $e->getType() !== $lastEnchantment->getType() &&
+						$e->getType()->isCompatibleWith($lastEnchantment->getType())
+				);
+
+				if(count($availableEnchantments) === 0){
+					break;
+				}
+
+				/** @var EnchantmentInstance $lastEnchantment */
+				$lastEnchantment = self::getRandomWeightedEnchantment($random, $availableEnchantments);
+				$resultEnchantments[] = $lastEnchantment;
+
+				$cost >>= 1;
+			}
+		}
+
+		$protocolEnchantments = array_map(
+			fn(EnchantmentInstance $e) => new Enchant(EnchantmentIdMap::getInstance()->toId($e->getType()), $e->getLevel()),
+			$resultEnchantments
+		);
+
+		return new EnchantOption($optionCost, $slot, $protocolEnchantments, [], [], self::getRandomOptionName($random), $slot);
+	}
+
+	private static function getEnchantability(Item $item) : int{
+		return match ($item::class) {
+			TieredTool::class => $item->getTier()->getEnchantability(),
+			Armor::class => $item->getMaterial()->getEnchantability(),
+			default => 1
+		};
+	}
+
+	/**
+	 * @return EnchantmentInstance[]
+	 */
+	private static function getAvailableEnchantments(int $cost, Item $item) : array{
+		$list = [];
+
+		foreach(VanillaEnchantments::getAll() as $enchantment){
+			if($enchantment->isTreasure()){
+				continue;
+			}
+			if(!$item instanceof Book && !$enchantment->hasPrimaryItemType($item->getEnchantmentFlag())){
+				continue;
+			}
+
+			for($lvl = $enchantment->getMaxLevel(); $lvl > 0; $lvl--){
+				if($cost >= $enchantment->getMinCost($lvl) && $cost <= $enchantment->getMaxCost($lvl)){
+					$list[] = new EnchantmentInstance($enchantment, $lvl);
+					break;
+				}
+			}
+		}
+
+		return $list;
+	}
+
+	/**
+	 * @param array<EnchantmentInstance> $enchantments
+	 */
+	private static function getRandomWeightedEnchantment(Random $random, array $enchantments) : ?EnchantmentInstance{
+		if(count($enchantments) === 0){
+			return null;
+		}
+
+		$totalWeight = 0;
+		foreach($enchantments as $enchantment){
+			$totalWeight += $enchantment->getType()->getRarity();
+		}
+
+		$result = null;
+		$randomWeight = $random->nextRange(1, $totalWeight);
+
+		foreach($enchantments as $enchantment){
+			$randomWeight -= $enchantment->getType()->getRarity();
+
+			if($randomWeight <= 0){
+				$result = $enchantment;
+				break;
+			}
+		}
+
+		return $result;
+	}
+
+	private static function getRandomOptionName(Random $random) : string{
+		$words = ['PocketMine', 'delights', 'players', 'enchanting', 'items', 'with', 'mystical', 'wonders'];
```

Comment:
These should be customizable

Reply:
How?

Reply:
Use your imagination

Reply:
Maybe generate a random string of English letters instead?